### PR TITLE
CLI for Chronolog Processing and Seperation from Frontent

### PR DESCRIPTION
**Pull request by ESummerscale**
Implemented CLI for processing, with documentation. 
Removed dependency on flask app for testing to fully separate concerns of frontend and backend.

### CODE REVIEW

File: chronolog.py [hunk 1]
```diff
@@ -0,0 +1,88 @@
+import os.path
+import regex as re
+
+import ChronologBackend.nlp as nlp
+import argparse
+from argparse import RawTextHelpFormatter
```
Comment (zeegtab) [inline chronolog.py:6]: Is it possible to just import what you need from argparse instead of importing all of it?

File: chronolog.py [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+DEFAULT_SPLITS = 10
+
```
Comment (zeegtab) [inline chronolog.py:9]: Would be useful to add some comments explaining what these regexes are, i.e what string patterns they match with.

File: chronolog.py [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+        "The analysis will be stored in {title}_analysis.json if this argument is given.")
+    parser.add_argument(
+        '--sections',
+        '-s',
+        required=False,
+        type=int,
+        metavar="N",
+        default=10,
+        help="Supply the number of sections you want to divide your book into. " +
+        "The timeline will then be created in N similar sized sections. Default N = 10.")
```
Comment (BOluokun) [inline chronolog.py:66]: (nit) Perhaps the `DEFAULT_SPLITS` constant from line 8 should be used here.

File: chronolog.py [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+                - digit             = Digit only
+                - numeral           = Numeral only
+                You may provide your own alternative chapter regex in the form of a regex string.
+                Please make sure your regex is well-formed to avoid undesirable results.
+                Note that in any user-defined regex, there must be NO capturing groups.
+                """
+    )
+    parser.add_argument('--quiet', '-q', required=False, action='store_true',
+                        help="Suppresses updates on progress of model.")
+    parser.add_argument(
```
Comment (zeegtab) [inline chronolog.py:50]: What happens if there are spaces in the novel's title? Are the spaces replaced with underscores?

File: scripts/regenerate_single_test_result.py [hunk 5]
```diff
@@ -4,7 +4,7 @@
 current = os.path.dirname(os.path.realpath(__file__))
 parent = os.path.dirname(current)
 sys.path.append(parent)
-from nlp import generate_interactions_matrix  # noqa: E402
+from ChronologBackend.nlp import generate_interactions_matrix  # noqa: E402
```
Comment (BOluokun) [inline scripts/regenerate_single_test_result.py:7]: (nit) the folder name should probably be lowercase.

File: ChronologBackend/nlp.py [hunk 6]
```diff
@@ -2,45 +2,58 @@
...
 
-def process_data(text, chapter_regex):
+def process_data(text, chapter_regex, num_splits, quiet):
     """
     :param text: The full text to be analysed.
     :param chapter_regex: Regex by which chapters are determined. If chapter_regex=="", split into equal length sections.
+    :param quiet: Flag to silence print statements
     :return: Returns list of sections
     """
+    if not num_splits:
```
Comment (BOluokun) [inline ChronologBackend/nlp.py:15]: In the chronolog main, we pass in `args.sections or DEFAULT_SPLITS` into the`num_splits` parameter so it should never be `None`. So I think we should remove this check.

**Review by BOluokun — COMMENTED** [review]
Overall good, I just have a few comments.

**Review by zeegtab — COMMENTED** [review]
Great stuff!

File: chronolog.py [hunk 7]
```diff
@@ -0,0 +1,88 @@
+import os.path
+import regex as re
+
+import ChronologBackend.nlp as nlp
+import argparse
+from argparse import RawTextHelpFormatter
+
+DEFAULT_SPLITS = 10
+
+NUMERAL_PATTERN = re.compile(
```
Comment (zeegtab) [inline chronolog.py:10]: Might be useful to have some comments explaining what these regexes do, i.e what kind of patterns do they match.

**Review by zeegtab — COMMENTED** [review]
(Changed placement of comments)

**Review by zeegtab — APPROVED** [review]
Great :)

**Comment by ESummerscale** [comment]
Partially fixes #76

**Pull request merged**